base-images-sbom-script/base_image_sbom_script.py

```diff
@@ -0,0 +1,86 @@
+import json
+import argparse
+import pathlib
+
+from packageurl import PackageURL
+
+
+def get_base_images_sbom_data(images, is_last_from_scratch):
+
+    new_data = {"formulation": [{"components": []}]}
+
+    for index, image in enumerate(images):
+        property_name = f"builder-{index}"
+        if index == len(images) - 1 and not is_last_from_scratch:
+           property_name = f"base-image"
+           
+           
+        repository, digest = image.split("@")
+        repository_without_tag, tag = repository.split(":")
+        # last fragment of the repository
+        name = repository_without_tag.split("/")[-1]
+        
+        purl = PackageURL(
+            type="oci",
+            name=name,
+            version=digest,
+            qualifiers={
+                "repository_url": repository_without_tag,
+                "tag": tag,
+            }
+        )
+        
+        component = {
+            "type": "container",
+            "name": repository_without_tag,
+            "purl": purl.to_string(),
+            "properties": [{
+                "name": property_name,
+                "value": index
+            }],
+        }
+        new_data["formulation"][0]["components"].append(component)
+
+
+
+    return new_data
+
+    
+def parse_args():
+    parser = argparse.ArgumentParser(description="Updates the sbom file with base images data based on the provided files")
+    parser.add_argument("--sbom", type=pathlib.Path, help="Path to the sbom file", required=True)
+    parser.add_argument("--base-images-raw", type=pathlib.Path, help="Path to the file containing raw base images", required=True)
+    parser.add_argument("--base-images-digests", type=pathlib.Path, help="Path to the file containing base images digests", required=True)
+    args = parser.parse_args()
+    
+    return args    
+    
+def main():
+
+    args = parse_args()
+    
+    with args.base_images_raw.open('r') as f:
+        raw_images = f.read().splitlines()
+        
+    with args.base_images_digests.open('r') as f:
+        digest_images = f.read().splitlines()
+    
+    is_last_from_scratch = False
+    if raw_images[-1] == "scratch":
+       is_last_from_scratch = True
+
+    
+    with args.sbom.open('r') as f:
+        sbom = json.load(f)
+        
+    new_data = get_base_images_sbom_data(digest_images, is_last_from_scratch)
+    sbom.update(new_data)
```

Comment:
If the SBOM already had a `formulation` attribute, this would overwrite its content completely. I think we just want to append a `{"components": ...}` item to the formulations

Reply:
Yeah, it make sense to use a better approach here. But formulation is an array, so if it would exist beforehand, then there is also a possibility to have more items in it. That might create additional problem and also, the individual formulation item could potentially have a "components" already. So how much should we prepare here for these kinds of options?

Reply:
Yeah you are right, there are actually no problems with this approach. Updated the code and added unit tests covering the scenarios where the input sbom did or did not have the formulation key